controllers/agents/svc/agentsvc_controller.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2023 The Primaza Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svc
+
+import (
+	"context"
+	"errors"
+
+	"github.com/primaza/primaza/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Agent Service Reconciler reconciles a Agent Service object
+type AgentServiceReconciler struct {
+	client.Client
+}
+
+func NewAgentServiceReconciler(mgr ctrl.Manager) *AgentServiceReconciler {
+	return &AgentServiceReconciler{
+		Client: mgr.GetClient(),
+	}
+}
+
+const agentsvcfinalizer = "agent.primaza.io/finalizer"
+
+func (r *AgentServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	l := log.FromContext(ctx)
+	l.Info("Reconcile Agent Service Deployment")
+
+	// first, get the agent svc deployment
+	agentsvcdeployment := appsv1.Deployment{}
+	objKey := client.ObjectKey{
+		Name:      "primaza-controller-agentsvc",
```

Comment:
I'd suggest to move that in a `const`. A name may be `ServiceAgentDeploymentName`. Also, a great place where to have that const would be `pkg/primaza/constants/constants.go`

```suggestion
		Name:      constants.ServiceAgentDeploymentName,
```

Reply:
Yeah that's better